### PR TITLE
Remove eunit leveled configuration

### PR DESCRIPTION
**Pull request by tolbrino**
- Add mnesia integration test configuration
- Make CT fail when mandatory env var isn't set

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -230,11 +218,6 @@ eunit-fortuna: KIND=test
 eunit-fortuna: internal-build
 	@ERL_FLAGS="-args_file $(EUNIT_VM_ARGS) -config $(EUNIT_SYS_CONFIG) -network_id local_fortuna_testnet" $(REBAR) do eunit $(EUNIT_TEST_FLAGS)
 
-eunit-mnesia-leveled: KIND=test
-eunit-mnesia-leveled: internal-build
```
Comment (lucafavatella) [inline Makefile]: I approve the removal of this target.
I am not convinced re the rest so not approving for now.

File: Makefile [hunk 2]
```diff
@@ -179,41 +179,29 @@ dialyzer: $(SWAGGER_ENDPOINTS_SPEC)
...
-		AETERNITY_TESTCONFIG_DB_BACKEND=leveled \
-			$(REBAR) ct $(CT_TEST_FLAGS) --sys_config config/test-mnesia-leveled.config; \
-	fi
+ct-mnesia-leveled: SYSCONFIG=config/test.config
+ct-mnesia-leveled: AETERNITY_TESTCONFIG_DB_BACKEND=leveled
+ct-mnesia-leveled: internal-ct
+
+ct-mnesia-rocksdb: KIND=test
+ct-mnesia-rocksdb: SYSCONFIG=config/test.config
+ct-mnesia-rocksdb: AETERNITY_TESTCONFIG_DB_BACKEND=rocksdb
```
Comment (lucafavatella) [inline Makefile]: I typed this as `AETERNITY_TESTCONFIG_DB_BACKEN` and I got:
```
*** User 2019-04-02 15:37:49.883 ***
Writing config (".../_build/test/logs/latest.aesc_fsm/dev1/data/aeternity.json")
{
  "chain": {
    "db_backend": "",
    "persist": false
  },
...
}
Validation failed
Position: chain/db_backend
Value   : ""
Schema  :
{
  "default": "unix:rocksdb|*:mnesia",
  "description": "Choice of database backend.",
  "example": "'rocksdb' OR 'unix:rocksdb' OR 'unix:rocksdb|*:mnesia",
  "pattern": "^([a-z]+|([a-z0-9\\*]+:[a-z]+)|([a-z0-9\\*]+:[a-z]+(\\h*\\|\\h*[a-z0-9\\*]+:[a-z]+))+)$",
  "type": "string"
}
Reason  : No match


*** User 2019-04-02 15:37:49.997 ***
Config (".../_build/test/logs/latest.aesc_fsm/dev1/data/aeternity.json") check: {error,
                                                                                                        validation_failed}

*** CT Error Notification 2019-04-02 15:37:49.997 ***
aesc_fsm_SUITE:init_per_suite failed
Reason: {badmatch,{error,validation_failed}}
```

File: Makefile [hunk 3]
```diff
@@ -407,6 +390,18 @@ internal-clean: $$(KIND)
...
 
+internal-ct: $$(KIND)
+internal-ct: $$(SYSCONFIG)
+internal-ct: $$(AETERNITY_TESTCONFIG_DB_BACKEND)
+internal-ct: internal-build
+	@NODE_PROCESSES="$$(ps -fea | grep bin/aeternity | grep -v grep)"; \
+	if [ $$(printf "%b" "$${NODE_PROCESSES}" | wc -l) -gt 0 ] ; then \
+		(printf "%b\n%b\n" "Failed testing: another node is already running" "$${NODE_PROCESSES}" >&2; exit 1);\
+	else \
+		AETERNITY_TESTCONFIG_DB_BACKEND=$(AETERNITY_TESTCONFIG_DB_BACKEND) \
```
Comment (lucafavatella) [inline Makefile]: I typed this as `AETERNITY_TESTCONFIG_DB_BACKEN=$(AETERNITY_TESTCONFIG_DB_BACKEND)` and I got:
```
*** CT Error Notification 2019-04-02 15:44:25.166 ***
aesc_fsm_SUITE:init_per_suite failed
Reason: {test_case_failed,The mandatory environment variable is not set: AETERNITY_TESTCONFIG_DB_BACKEND}
```

File: Makefile [hunk 4]
```diff
@@ -179,41 +179,29 @@ dialyzer: $(SWAGGER_ENDPOINTS_SPEC)
...
-ct-mnesia-leveled: internal-build
-	@NODE_PROCESSES="$$(ps -fea | grep bin/aeternity | grep -v grep)"; \
-	if [ $$(printf "%b" "$${NODE_PROCESSES}" | wc -l) -gt 0 ] ; then \
-		(printf "%b\n%b\n" "Failed testing: another node is already running" "$${NODE_PROCESSES}" >&2; exit 1);\
-	else \
-		AETERNITY_TESTCONFIG_DB_BACKEND=leveled \
-			$(REBAR) ct $(CT_TEST_FLAGS) --sys_config config/test-mnesia-leveled.config; \
-	fi
+ct-mnesia-leveled: SYSCONFIG=config/test.config
+ct-mnesia-leveled: AETERNITY_TESTCONFIG_DB_BACKEND=leveled
```
Comment (lucafavatella) [inline Makefile]: I understand this is target-specific variable.

File: Makefile [hunk 5]
```diff
@@ -407,6 +390,18 @@ internal-clean: $$(KIND)
 internal-distclean: $$(KIND)
 	@rm -rf ./_build/$(KIND)
 
+internal-ct: $$(KIND)
+internal-ct: $$(SYSCONFIG)
+internal-ct: $$(AETERNITY_TESTCONFIG_DB_BACKEND)
```
Comment (lucafavatella) [inline Makefile]: Is this a prerequisite on a target named e.g. leveled?

Comment (tolbrino) [reply Makefile]: No. What would that target do?

**Comment by tolbrino** [comment]
@uwiger @ThomasArts Can you review this?

@lucafavatella Is this good to go from your end?

File: Makefile (hunk 5 above)
Comment (lucafavatella) [reply Makefile]: I spoke with @tolbrino , this line seems unnecessary.

File: Makefile [hunk 6]
```diff
@@ -378,35 +361,41 @@ multi-build: dev1-build
...
-internal-compile-deps: $$(KIND)
+internal-compile-deps:
 	@$(REBAR) as $(KIND) compile -d
 
-internal-package: $$(KIND)
 internal-package: REVISION internal-compile-deps $(SWAGGER_ENDPOINTS_SPEC)
 	@$(REBAR) as $(KIND) tar
 
-internal-build: $$(KIND)
+internal-build:
```
Comment (lucafavatella) [inline Makefile]:
```suggestion
```

**Pull request merged**